### PR TITLE
feat(dialog): add new `<bq-dialog>` component

### PR DESCRIPTION
**Pull request by endv-bogdanb**

### CODE REVIEW

File: packages/bee-q/src/components/dialog/bq-dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  // State() variables
+  @State() isOpen = false;
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  // Public Property API
+  // ========================
+
+  /** The size of the dialog */
+  @Prop({ reflect: true, mutable: true }) size: TDialogSize = 'large';
```
Comment (dgonzalezr) [inline packages/bee-q/src/components/dialog/bq-dialog.tsx:36]: I think the default value in terms of size for any component is `medium`.

File: packages/bee-q/src/components/dialog/bq-dialog.types.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export const DIALOG_SIZE = ['small', 'medium', 'large'] as const;
+export type TDialogSize = (typeof DIALOG_SIZE)[number];
+
+export const DIALOG_FOOTER_VARIANT = ['standard', 'light'] as const;
```
Comment (dgonzalezr) [inline packages/bee-q/src/components/dialog/bq-dialog.types.ts:4]: I would recommend using `highlight` as a value for the footer appearance when it has a background color gray.

```suggestion
export const DIALOG_FOOTER_VARIANT = ['standard', 'highlight'] as const;
```

File: packages/bee-q/src/components/dialog/bq-dialog.tsx (hunk 1 above)
Comment (endv-bogdanb) [reply packages/bee-q/src/components/dialog/bq-dialog.tsx]: Done

File: packages/bee-q/src/components/dialog/bq-dialog.types.ts (hunk 2 above)
Comment (endv-bogdanb) [reply packages/bee-q/src/components/dialog/bq-dialog.types.ts]: Done

File: packages/bee-q/src/components/dialog/bq-dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+  /** The variant of button to apply on top of the appearance */
+  @Prop({ reflect: true }) variant: TDialogFooterVariant = 'standard';
```
[the remark this answers is not in the document]
Comment (endv-bogdanb) [reply packages/bee-q/src/components/dialog/bq-dialog.tsx]: Done

File: packages/bee-q/src/components/dialog/scss/bq-dialog.scss [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  @apply visible #{!important};
+}
+
+/* --------------------------------- Dialog --------------------------------- */
+
+.bq-dialog {
+  @apply m-auto flex flex-col gap-l p-0;
+  @apply z-[var(--bq-dialog-z-index)] rounded-[var(--bq-dialog--border-radius)] border-none bg-[var(--bq-dialog--background)] shadow-[shadow:var(--bq-dialog--box-shadow)];
+  @apply opacity-50 transition-opacity duration-300 ease-in-out;
+  // transform: scale(0);
```
Comment (endv-bogdanb) [inline packages/bee-q/src/components/dialog/scss/bq-dialog.scss:23]:
```suggestion
```

**Review by dgonzalezr — APPROVED** [review]

**Pull request merged**